misc/services/camel/de-metas-camel-externalsystems/de-metas-camel-scriptedadapter/javascript_templates/d2m_so.js

```diff
@@ -0,0 +1,94 @@
+/*
+ * #%L
+ * de-metas-camel-scriptedadapter
+ * %%
+ * Copyright (C) 2025 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+function transform(messageToMetasfresh) {
+    try {
+        const input = JSON.parse(messageToMetasfresh);
+
+        if (!Array.isArray(input.d2m_so)) {
+            throw new Error("Input JSON must contain a 'd2m_so' array.");
+        }
+
+        const results = [];
+        input.d2m_so.forEach(so => {
+                if (Array.isArray(so.lines)) {
```

Comment:
i'm not into JS, but maybe there should be a null-check for so.lines, so.datePromised and so.dateOrdered